src/transfer-manager.ts

```diff
@@ -0,0 +1,433 @@
+    options: LargeFileDownloadOptions,
+    callback: DownloadCallback
+  ): Promise<void>;
+  /**
+   * @typedef {object} LargeFileDownloadOptions
+   * @property {number} [concurrencyLimit] The number of concurrently executing promises
+   * to use when downloading the file.
+   * @property {number} [chunkSizeBytes] The size in bytes of each chunk to be downloaded.
+   * @experimental
+   */
+  /**
+   * Download a large file in chunks utilizing parallel download operations. This is a convenience method
+   * that utilizes {@link File#download} to perform the download.
+   *
+   * @param {object} [file] {@link File} to download.
+   * @param {LargeFileDownloadOptions} [options] Configuration options.
+   * @param {DownloadCallback} [callbac] Callback function.
+   * @returns {Promise<DownloadResponse | void>}
+   *
+   * @example
+   * ```
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const bucket = storage.bucket('my-bucket');
+   * const transferManager = new TransferManager(bucket);
+   *
+   * //-
+   * // Download a large file in chunks utilizing parallel operations.
+   * //-
+   * transferManager.downloadLargeFile(bucket.file('large-file.txt'), function(err, contents) {
+   * // Your local directory now contains:
+   * // - "large-file.txt" (with the contents from my-bucket.large-file.txt)
+   * });
+   *
+   * //-
+   * // If the callback is omitted, we will return a promise.
+   * //-
+   * const response = await transferManager.downloadLargeFile(bucket.file('large-file.txt');
+   * @experimental
+   */
+  async downloadLargeFile(
+    file: File,
+    optionsOrCallback?: LargeFileDownloadOptions | DownloadCallback,
+    callback?: DownloadCallback
+  ): Promise<void | DownloadResponse> {
+    let options: LargeFileDownloadOptions = {};
+    if (typeof optionsOrCallback === 'function') {
+      callback = optionsOrCallback;
+    } else if (optionsOrCallback) {
+      options = optionsOrCallback;
+    }
+
+    let chunkSize = options.chunkSizeBytes || LARGE_FILE_DEFAULT_CHUNK_SIZE;
+    let limit = pLimit(
+      options.concurrencyLimit || DEFAULT_PARALLEL_LARGE_FILE_DOWNLOAD_LIMIT
+    );
+    const promises = [];
+
+    const fileInfo = await file.get();
+    const size = parseInt(fileInfo[0].metadata.size);
+    // If the file size does not meet the threshold download it as a single chunk.
+    if (size < LARGE_FILE_SIZE_THRESHOLD) {
+      limit = pLimit(1);
+      chunkSize = size;
+    }
+
+    let start = 0;
+    const fileToWrite = await fsp.open(path.basename(file.name), 'w+');
+    while (start < size) {
+      const chunkStart = start;
+      let chunkEnd = start + chunkSize - 1;
+      chunkEnd = chunkEnd > size ? size : chunkEnd;
+      promises.push(
+        limit(() =>
+          file.download({start: chunkStart, end: chunkEnd}).then(resp => {
+            return fileToWrite.write(resp[0], 0, resp[0].length, chunkStart);
+          })
+        )
+      );
```

Comment:
In case the first chunk fails, we could lazily create the file like we do here:
https://github.com/googleapis/nodejs-storage/blob/a61e6194c50fa2fb8d22d37626a62c5352275850/src/file.ts#L2156-L2174